src/cryptonote_core/service_node_voting.cpp

```diff
@@ -298,6 +317,35 @@ namespace service_nodes
     return result;
   }
 
+  quorum_vote_t make_checkpointing_vote(crypto::hash const &block_hash, uint64_t block_height, uint16_t index_in_quorum, crypto::public_key const &pub_key, crypto::secret_key const &sec_key)
+  {
+    quorum_vote_t result         = {};
+    result.type                  = quorum_type::checkpointing;
+    result.checkpoint.block_hash = block_hash;
+    result.block_height          = block_height;
+    result.group                 = quorum_group::worker;
+    result.index_in_group        = index_in_quorum;
+    result.signature             = make_signature_from_vote(result, pub_key, sec_key);
+    return result;
+  }
+
+  cryptonote::checkpoint_t make_empty_service_node_checkpoint(crypto::hash const &block_hash, uint64_t height)
+  {
+    cryptonote::checkpoint_t result = {};
+    result.type                     = cryptonote::checkpoint_type::service_node;
+    result.height                   = height;
+    result.block_hash               = block_hash;
+    return result;
+  }
+
+  voter_to_signature vote_to_voter_to_signature(quorum_vote_t const &vote)
+  {
+    voter_to_signature result = {};
+    result.voter_index        = vote.index_in_group;
+    result.signature          = vote.signature;
+    return result;
+  }
```

Comment:
Is there any reason not to implement these three factory functions as constructors of the respective types?

Reply:
I've removed the `vote_to_voter_to_signature()` for a constructor.

Making checkpointing vote/state change vote are mostly the same except one takes a block hash, the other takes a state to vote into. I prefer essentially, a named constructor for each vs 2 constructors with slightly different arguments for readability.

The making service node checkpoint is also to emphasise the empty part. Feels weird semantically having one of the constructors making essentially an invalid service node checkpoint because it doesn't specify signatures. Making it named makes it obvious its not a properly constructed checkpoint.